state/machine.go

```diff
@@ -2322,6 +2379,21 @@ func (m *Machine) UpdateOperation() *UpdateMachineOperation {
 	return &UpdateMachineOperation{m: &Machine{st: m.st, doc: m.doc}}
 }
 
+func (m *Machine) getUpgradeSeriesLock() (*upgradeSeriesLockDoc, error) {
```

Comment:
Can we have a new struct here with the data, rather than the Doc struct, to return here? upgradeSeriesLock? Looking at the pattern set with getInstanceData()

Reply:
getMachineDoc doesn't do this. it gets the doc directly. Is what getInstanceData() doing a pattern or just a one off?

Reply:
I haven't checked beyond the state/machine.  A Doc seems only be directly accessed when changing it.  Getting data from a doc is returned in a single type, or a struct not of the Doc type.

Reply:
I don't see the pattern. I see some methods in the package and other packages which do things like return entire raw collections. I don't see an issue with these since it they are unexported methods. None of the exported methods return the entire document directly.